metals/src/main/scala/scala/meta/internal/decorations/SemanticdbTreePrinter.scala

```diff
@@ -123,16 +124,48 @@ object SemanticdbTreePrinter {
        */
       case tree @ s.ApplyTree(_: s.OriginalTree, _)
           if userConfig.showImplicitArguments =>
-        printTree(tree)
+        for {
+          syntheticString <- printTree(tree).toList
+          range <- synthetic.range.toList
+        } yield (syntheticString, range)
 
       /**
        *  def hello[T](T object) = object
        *  hello<<[String]>>("")
        */
       case tree @ s.TypeApplyTree(_: s.OriginalTree, _)
           if userConfig.showInferredType =>
-        printTree(tree)
-      case _ => None
+        for {
+          syntheticString <- printTree(tree).toList
+          range <- synthetic.range.toList
+        } yield (syntheticString, range)
+      /**
+       *  implicit def implicitFun(object: T): R = ???
+       *  def fun(r: R) = ???
+       *  fun(<<implicitFun(>>new T<<)>>)
+       */
+      case s.ApplyTree(id: s.IdTree, _) if userConfig.showImplicitConversions =>
+        def synthetics(syntheticString: String, range: s.Range) = {
+          if (isHover && !isFullLine)
+            List(
+              (syntheticString, range.withEndCharacter(range.startCharacter))
+            )
+          else
+            List(
+              (
+                syntheticString + "(",
+                range.withEndCharacter(range.startCharacter)
+              ),
+              (")", range)
+            )
```

Comment:
So I'm not fully sure I understand this. Why is the extra `( )` needed in the case that it's a hover request and a `isFullLine` here`?

Reply:
Ach, that is pretty unclear, I will add a comment. That's just for making the line look like:
`scala.Predef.augmentString("foo").map(c => scala.LowPriorityImplicits.charWrapper(c).toUpper)`

instead of:

`scala.Predef.augmentString"foo".map(c => scala.LowPriorityImplicits.charWrapper(c).toUpper)`

for the fallback hover.